src/transformStateWithClones.js

```diff
@@ -65,7 +65,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const stateHistory = [];
+  const clonedState = { ...state };
+
+  transforms.forEach(element => {
+    const { operation, properties } = element;
+
+    if (operation === 'addProperties') {
+      for (const key in properties) {
+        clonedState[key] = properties[key];
+      }
+    }
+
+    if (operation === 'removeProperties') {
```

Comment:
```suggestion
    if (operation === REMOVE) {
```
Добре виносити літерали в константи, щоб потім перевикористовувати